azkaban-common/src/test/java/azkaban/jobtype/JobTypeManagerWithDynamicClusterTest.java

```diff
@@ -267,4 +269,60 @@ public void testJobRunnerObservesClusterSpecificParams() {
     jobProps = PropsUtils.resolveProps(jobProps);
     Assert.assertEquals("valB", jobProps.getString("PropA"));
   }
+
+  /**
+   * Verify that overidden plugin load props are indeed overridden.
+   */
+  @Test
+  public void testPluginLoadPropsOverrideByClusterSpecificParams() {
+    final JobTypeManager manager = new JobTypeManager(this.testPluginDirPath, null,
+        this.getClass().getClassLoader(), this.clusterRouter, null);
+
+    Props jobProps = new Props();
+    jobProps.put("type", "testjob");
+    jobProps.put(PIG_HOME, "/user/pig/path");
+    jobProps.put(CommonJobProperties.JOB_CLUSTER_COMPONENTS_DEPENDENCIES, "hadoop");
+
+    final JobParams jobParams = manager.createJobParams("testjob", jobProps, LOG);
+    Assert.assertTrue(
+        jobParams.contextClassLoader instanceof HadoopSecurityManagerClassLoader);
+
+    final Job job = JobTypeManager.createJob("testjob", jobParams, LOG);
+    Props sysProps = ((FakeJavaJob2) job).getSysProps();
+    Assert.assertEquals(null, sysProps.get("PropA"));
+
+    // Create JobTypeManager again but with pluginOverrideProps this time.
+    final String pluginLoadOverrideProps = "PropC,PropB";
```

Comment:
What was the value in system props for this? Sorry I am missing something.

Reply:
System props does not have it. However, it is supposed to be populated as the two props exist in cluster props.